# Claude Code review — [AAASM-2097]

## CI state

**25/25 SUCCESS** across the full node-sdk CI matrix — napi-build × 4 (ubuntu/macos/windows × Node 20+22), module-smoke (18/20/22), test (18/20/22/24 × ubuntu/macos/windows = 12 entries), quality, coverage-and-analysis, SonarCloud, codecov/patch. `mergeable=MERGEABLE`, `mergeStateStatus=CLEAN`.

The CI runs `pnpm install` with the existing lockfile (without alpha-1 lockfile drift, since that's a separate PR #57). All publish-step changes are workflow-only — they execute on tag push, not on PR CI. Verification closes when the next tag pushes.

## Scope vs. acceptance criteria

| AC | Where verified | Status |
|---|---|---|
| Pushing a pre-release tag publishes `@agent-assembly/sdk` under the `@alpha` dist-tag | `pnpm publish ... --tag alpha` resolved at runtime based on `node -p "require('./package.json').version"` containing `-` | ✅ logic verified by inspection |
| Pushing a GA tag publishes under `@latest` as before | `DIST_TAG=""` branch when version doesn't contain `-`, falls back to npm default | ✅ |
| Applied to **all** publish surfaces in the repo | Diff touches 3 distinct steps: release.yml line 38, release-node.yml line 159 (4-runtime loop), release-node.yml line 167 (main SDK) | ✅ |
| `actionlint` clean on both modified files | Verified locally before push | ✅ |
| `npm view @agent-assembly/sdk dist-tags` shows both `alpha` and `latest` after both release types | Cannot fully verify until next pre-release tag publishes — deferred | ⏳ deferred to live verification |

## Implementation detail worth recording

- **Shell-time computation vs. expression-time**: `VERSION=$(node -p "require('./package.json').version")` runs inside the bash step rather than via GitHub Actions expressions. This avoids needing a separate `Compute dist-tag` step and keeps the publish step self-contained. The cost is a small bash if-block per step.
- **`# shellcheck disable=SC2086`**: intentional. `$DIST_TAG` either expands to `--tag alpha` (2 tokens) or `` (empty), so we want word-splitting. Quoting would break it.
- **Future `-beta.*` / `-rc.*` work**: any version with `-` currently maps to `alpha`. When -beta/-rc need their own dist-tags, the if-branch can be expanded to detect the identifier substring (`*-beta.*` → `beta`, etc.). Out of scope for this PR — the alpha-1 dry-run is the only pre-release on the immediate roadmap.

## Verdict

**Ready for human approval and merge.** Full CI green, scope-AC items all verified by inspection or CI, the deferred "live verification" item is the appropriate-by-construction outcome for a publish-time-only change.

— Claude Code (Opus 4.7, 1M context)


[AAASM-2097]: https://lightning-dust-mite.atlassian.net/browse/AAASM-2097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

--- a/.github/workflows/release-node.yml
+++ b/.github/workflows/release-node.yml
@@ -155,8 +155,10 @@ jobs:
         run: |
           set -euo pipefail
           VERSION=$(node -p "require('./package.json').version")
-          if [[ "$VERSION" == *-* ]]; then
-            DIST_TAG="--tag alpha"
+          # Derive the npm dist-tag from the SemVer pre-release identifier.
+          # 0.0.1-alpha.1 → --tag alpha, 0.0.1-rc.1 → --tag rc, 0.0.1 → (empty, defaults to @latest)
+          if [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+-([a-zA-Z]+) ]]; then
+            DIST_TAG="--tag ${BASH_REMATCH[1]}"
           else
             DIST_TAG=""
           fi
@@ -174,8 +176,10 @@ jobs:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         run: |
           VERSION=$(node -p "require('./package.json').version")
-          if [[ "$VERSION" == *-* ]]; then
-            DIST_TAG="--tag alpha"
+          # Derive the npm dist-tag from the SemVer pre-release identifier.
+          # 0.0.1-alpha.1 → --tag alpha, 0.0.1-rc.1 → --tag rc, 0.0.1 → (empty, defaults to @latest)
+          if [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+-([a-zA-Z]+) ]]; then
+            DIST_TAG="--tag ${BASH_REMATCH[1]}"
           else
             DIST_TAG=""
           fi

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -39,8 +39,10 @@ jobs:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         run: |
           VERSION=$(node -p "require('./package.json').version")
-          if [[ "$VERSION" == *-* ]]; then
-            DIST_TAG="--tag alpha"
+          # Derive the npm dist-tag from the SemVer pre-release identifier.
+          # 0.0.1-alpha.1 → --tag alpha, 0.0.1-rc.1 → --tag rc, 0.0.1 → (empty, defaults to @latest)
+          if [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+-([a-zA-Z]+) ]]; then
+            DIST_TAG="--tag ${BASH_REMATCH[1]}"
           else
             DIST_TAG=""
           fi

